### ISSUE TITLE
ParamConverter not working for property

### ISSUE THREAD

**Issue by htesligte** [sensiolabs/SensioFrameworkExtraBundle#400]
Hi, 

https://github.com/sensiolabs/SensioFrameworkExtraBundle/blob/master/Request/ParamConverter/DoctrineParamConverter.php#L54
Isn't this check supposed to be "if (null === $object"?

DoctrineParamConverter::find returns null on NoResultException and the default repository also returns null when no results are found. Why is it checking for false?

**Comment by jakzal** [comment]
`DoctrineParamConverter::find()` returns `false` when [it realises](https://github.com/sensiolabs/SensioFrameworkExtraBundle/blob/master/Request/ParamConverter/DoctrineParamConverter.php#L76-L84) it cannot perform "find by identifier" lookup. The same happens with the criteria lookup in `DoctrineParamConverter::findOneBy()`. 

 `null` comparision is made in [the condition a bit further down](https://github.com/sensiolabs/SensioFrameworkExtraBundle/blob/master/Request/ParamConverter/DoctrineParamConverter.php#L65-L67). 

In other words both methods can return either `false`, `null` or an `object`.

**Comment by htesligte** [comment]
I ran into this issue with the FOSRestBundle that generates routes automatically, but the same behaviour without the FOSRestBundle can be reproduced with this function definition:
```
/**
* @Route("/user/{guid}")
*/
public function showUserAction( User $guid ) // User $user works properly
```
The URL /user/<guid> causes getIdentifier to return the guid specified in the URL. However, the guid is not the primary key in the database. So DoctrineParamConverter::find tries to call the entity repository with find(<guid>) which returns null, and so DoctrineParamConverter::findOneBy is never called. 

I worked around this issue by creating my own converter and calling that before DoctrineParamConverter, but I wasn't sure if this is intended behaviour..

**Comment by mdeboer** [comment]
@fabpot can't this be closed as there is a mapping option now? Using the mapping option works fine for me in these cases.
